flow/project.py

```diff
@@ -1976,11 +2224,17 @@ def submit(self, bundle_size=1, jobs=None, names=None, num=None, parallel=False,
             bool
         :param walltime:
             Specify the walltime in hours or as instance of datetime.timedelta.
+        :param mode:
+            Defines the execution mode. 'run' will perform operations using the
+            run command's logic on the compute nodes. 'exec' will perform all
+            pre and post checks on the submit side. Default is 'run'.
+        :type mode:
+            str
         """
         # Regular argument checks and expansion
         if jobs is None:
             jobs = self  # select all jobs
-        if isinstance(names, str):
+        if isinstance(names, six.string_types):
```

Comment:
please revert this change

Reply:
You resolved this conversation, but did not make the change. That makes it extremely hard for me to keep track of things. Please do not resolve conversations before you have resolved them. Thx!